src/Engine/GameEvents/GameEventListener.java

```diff
@@ -71,12 +74,12 @@ public final void unregister()
   public void receiveCreateUnitEvent(Unit unit){};
   public void receiveCaptureEvent(Unit unit, Location location){};
   public void receiveCommanderDefeatEvent(CommanderDefeatEvent event){};
-  public void receiveLoadEvent(LoadEvent event){};
+  public void receiveLoadEvent(LoadLifecycle.LoadEvent event){};
   public void receiveMoveEvent(MoveEvent event){};
-  public void receiveUnitJoinEvent(UnitJoinEvent event){};
+  public void receiveUnitJoinEvent(JoinLifecycle.JoinEvent event){};
   public void receiveResupplyEvent(ResupplyEvent event){};
   public void receiveUnitDieEvent(UnitDieEvent event){};
-  public void receiveUnloadEvent(UnloadEvent event){};
+  public void receiveUnloadEvent(UnloadLifecycle.UnloadEvent event){};
```

Comment:
Why do you use <action>`Lifecycle for some but not others?

Also, these functions probably shouldn't actually take the event anyway; they should instead accept the relevant components (like Battle, Capture, MassDamage, etc).

Reply:
I dunno, it's how it turned out when I was moving things around and renaming.
Some of them seemed to require it, others didn't.

Guess I should try to standardize it?

Reply:
Maybe. I generally change the listener functions to accept the necessary components rather than the whole event once I actually use it somewhere, so it may make sense to just do that up front.

I didn't bother before though, so it's clearly not a priority.